packages/conjure-client/src/fetchBridge/fetchBridge.ts

```diff
@@ -61,9 +61,23 @@ export class FetchBridge implements IHttpApiBridge {
         this.getBaseUrl = typeof params.baseUrl === "function" ? params.baseUrl : () => params.baseUrl as string;
         this.getToken = typeof params.token === "function" ? params.token : () => params.token as string | undefined;
         this.fetch = params.fetch;
-        this.userAgent = new UserAgent(params.userAgent, [
-            { productName: "conjure-typescript-runtime", productVersion: IMPLEMENTATION_VERSION },
-        ]);
+
+        const additionalUserAgent = {
+            productName: "conjure-typescript-runtime",
+            productVersion: IMPLEMENTATION_VERSION,
+        };
+        if (Array.isArray(params.userAgent)) {
+            const userAgents = params.userAgent;
+            if (userAgents.length === 0) {
+                throw new Error("Can not construct bridge: no user agents provided");
+            }
+            const primaryUserAgent = userAgents[0];
+            const informationUserAgents = userAgents.slice(1).concat(additionalUserAgent);
+
+            this.userAgent = new UserAgent(primaryUserAgent, informationUserAgents);
+        } else {
+            this.userAgent = new UserAgent(params.userAgent, [additionalUserAgent]);
+        }
```

Comment:
Lets simplify things by changing the signature of the `UserAgent` method.

```suggestion
        const userAgents = Array.isArray(params.userAgent) ? params.userAgent : [params.userAgent];
        if (userAgents.length === 0) {
            throw new Error("At least one user agent must be provided");
        }
        this.userAgent = new UserAgent([...userAgents, CONJURE_USER_AGENT]);
```

Reply:
Fair enough, can do that.